main.c

```diff
@@ -97,17 +99,36 @@ static void handle_active_selection_motion(struct slurp_seat *seat, struct slurp
 	int32_t anchor_y = current_selection->anchor_y;
 	int32_t dist_x = current_selection->x - anchor_x;
 	int32_t dist_y = current_selection->y - anchor_y;
+    int32_t start_x = current_selection->start_x;
+    int32_t start_y = current_selection->start_y;
 
 	current_selection->has_selection = true;
-	// selection includes the seat and anchor positions
-	int32_t width = abs(dist_x) + 1;
-	int32_t height = abs(dist_y) + 1;
-	if (seat->state->aspect_ratio) {
+
+	int32_t width;
+	int32_t height;
+
+    if (seat->state->mirror_mode) {
+        width  = 2 * abs(start_x - current_selection->x);
+        height = 2 * abs(start_y - current_selection->y);
+    } else {
+        // selection includes the seat and anchor positions
+        width = abs(dist_x) + 1;
+        height = abs(dist_y) + 1;
+    }
+
+    if (seat->state->aspect_ratio) {
 		width = max(width, height / seat->state->aspect_ratio);
 		height = max(height, width * seat->state->aspect_ratio);
 	}
-	current_selection->selection.x = dist_x > 0 ? anchor_x : anchor_x - (width - 1);
-	current_selection->selection.y = dist_y > 0 ? anchor_y : anchor_y - (height - 1);
+    if (seat->state->mirror_mode) {
+        int32_t delta_x = width / 2;
+        int32_t delta_y = height / 2;
+        anchor_x = start_x - delta_x;
+        anchor_y = start_y - delta_y;
+    }
+
+	current_selection->selection.x = dist_x > 0 || seat->state->mirror_mode ? anchor_x : anchor_x - (width - 1);
+	current_selection->selection.y = dist_y > 0 || seat->state->mirror_mode ? anchor_y : anchor_y - (height - 1);
```

Comment:
Again, the -1 offset when dist is negative may be required for mirror mode. We may need to structure the code a bit differently to write the code only once.

PS: is the -1 to compensate the border? in that case we need to use state border_weight instead.